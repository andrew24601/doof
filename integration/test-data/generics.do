// Integration test for generic functions and classes

function identity<T>(value: T): T {
    return value;
}

class Box<T> {
    value: T;

    getValue(): T {
        return this.value;
    }
}

function main(): int {
    let number: int = identity<int>(7);
    let label: string = identity<string>("generic");

    let boxedNumber: Box<int> = Box<int> { value: number };
    let boxedLabel: Box<string> = Box<string> { value: label + "s" };

    println(number);
    println(label);
    println(boxedNumber.getValue());
    println(boxedLabel.getValue());

    return 0;
}
