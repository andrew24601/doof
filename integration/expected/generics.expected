7
generic
7
generics
